scripts/reference_disks/create_images.sh

```diff
@@ -0,0 +1,390 @@
+    maybe_run sudo mkfs.ext4 -m 0 -E lazy_itable_init=0,lazy_journal_init=0,discard "/dev/disk/by-id/google-$image_name"
+
+    # copied from https://cloud.google.com/compute/docs/disks/add-persistent-disk#mounting
+    maybe_run sudo mkdir -p "/mnt/disks/$image_name"
+    maybe_run sudo mount -o discard,defaults "/dev/disk/by-id/google-$image_name" "/mnt/disks/$image_name"
+}
+
+# Download a file to the mounted persistent disk
+download_file() {
+    local image_name
+    local cloud_path
+    local local_path
+
+    image_name="$1"
+    cloud_path="gs://$BUCKET/$2"
+    local_path="${cloud_path#"gs://"}"
+
+    maybe_run sudo gsutil -m cp "$cloud_path" "/mnt/disks/$image_name/$local_path"
+}
+
+# Unmount the populated disk, copy the disk to a public image, then delete the disk
+convert_disk_to_image() {
+    local image_name
+
+    image_name="$1"
+
+    maybe_run sudo sync
+
+    maybe_run sudo umount "/mnt/disks/$image_name"
+
+    maybe_run sudo rmdir "/mnt/disks/$image_name"
+
+    maybe_run gcloud compute instances detach-disk "$(hostname)" --disk "$image_name"
+
+    maybe_run gcloud compute images create "$image_name" --source-disk "$image_name"
+
+    # If you get an 403 error here then someone has shut off compute.images.setIamPolicy for the default compute SA.
+    # Its then best to find out why they reverted those permissions and then consider running this script as another SA.
+    # https://console.cloud.google.com/iam-admin/roles/details/projects%3Cbroad-dsde-cromwell-dev%3Croles%3CCustomComputeImagesAdmin?project=broad-dsde-cromwell-dev
+    maybe_run gcloud compute images add-iam-policy-binding "$image_name" \
+        --member "allAuthenticatedUsers" --role "roles/compute.imageUser"
+
+    maybe_run gcloud --quiet compute disks delete "$image_name"
+}
+
+# Check that a dependent program is available
+check_program() {
+    local program
+
+    program="$1"
+    if ! command -v "$program" &>/dev/null; then
+        error_msg "Program not found: \`$program\`"
+        MISSING_DEPENDENCIES=true
+    fi
+}
+
+# Check that a dependent file is readable
+check_file() {
+    local file
+
+    file="$1"
+    if [[ ! -r "$file" ]]; then
+        error_msg "File not found: '$file'"
+        MISSING_DEPENDENCIES=true
+    fi
+}
+
+create_images() {
+    local image_tsv
+    local image_name_prefixes
+    local image_name_prefix
+    local image_name
+    local image_manifest
+    local content_size
+    local image_size
+    local cloud_objects
+    local cloud_object
+
+    image_tsv="$1"
+
+    # Set the default zone for future gcloud commands to this machine's zone
+    CLOUDSDK_COMPUTE_ZONE=$(
+        gcloud compute instances list --filter "name=('$(hostname)')" --format "value(zone)"
+    )
+    export CLOUDSDK_COMPUTE_ZONE
+
+    # Get a list of image name prefixes, excluding 'none'
+    image_name_prefixes=$(cut -f 3 "$image_tsv" | grep -v '^none$' | sort | uniq)
```

Comment:
TOL any reason not to `-u` the `sort`?
```suggestion
    image_name_prefixes=$(cut -f 3 "$image_tsv" | grep -v '^none$' | sort -u)
```

Reply:
Force of habit / muscle memory. I'm often running `sort | uniq -c` to get counts.